robottelo/rerun_failures/uncollector.py

```diff
@@ -0,0 +1,111 @@
+        )
+    return launch
+
+
+def _get_failed_tests(launch, fail_args):
+    """Returns failed test names from Report Portal Launch based on arguments"""
+    tests = []
+    if fail_args == ['all']:
+        test_names = [*launch.tests(status='failed').keys()]
+        tests.extend(test_names)
+    else:
+        for dtype in fail_args:
+            test_names = [*launch.tests(defect_type=dtype).keys()]
+            tests.extend(test_names)
+    # Formating test names for 'member of items' operation
+    tests = [str(test).replace('::', '.') for test in tests]
+    return tests
+
+
+def _get_test_collection(rp_failed_tests, items):
+    """Returns the selected and deselected items"""
+    # Select test item if its in failed tests else deselect
+    LOGGER.debug('Selecting/Deselecting tests based on latest launch test results..')
+    selected = []
+    deselected = []
+    for item in items:
+        test_item = f"{item.location[0]}.{item.location[2]}"
+        if test_item in rp_failed_tests:
+            selected.append(item)
+        else:
+            deselected.append(item)
+    return selected, deselected
+
+
+def pytest_addoption(parser):
+    """Add options for pytest to collect only failed tests"""
+    help_script = '''
+        Collects only failed tests in Report Portal to run only failed tests.
+
+        Usage: --only-failed [options]
+
+        Options: [ specific_defect_type | comma_sparated_list_of defect_types ]
+
+        Defect_types:
+        product_bug: Rerun failed tests marked with product_bug on RP
+        rhel_bug: Rerun failed tests marked with rhel_bug on RP
+        satellite_bug: Rerun failed tests marked with rhel_bug on RP
+        automation_bug': Rerun failed tests marked with automation_bug on RP
+        robottelo_bug': Rerun failed tests marked with robottelo_bug on RP
+        nailgun_bug': Rerun failed tests marked with nailgun_bug on RP
+        airgun_bug': Rerun failed tests marked with airgun_bug on RP
+        system_issue': Rerun failed tests marked with system_issue on RP
+        saucelabs_issue': Rerun failed tests marked with saucelabs_issue on RP
+        to_investigate': Rerun failed tests marked with to_investigate on RP
+        no_defect': Rerun failed tests marked with no_defect on RP
+        '''
+    parser.addoption("--only-failed", const='all', nargs='?', help=help_script)
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_collection_modifyitems(items, config):
+    """
+    Collects tests based on pytest option to select tests marked as failed on Report Portal
+    """
+    fail_args = config.getvalue('only_failed')
+    if not fail_args:
+        return
+    rp = ReportPortal()
+    fail_args = fail_args.split(',') if ',' in fail_args else [fail_args]
+    allowed_args = [*rp.defect_types.keys()]
+    if not fail_args == ['all']:
+        if not set(fail_args).issubset(set(allowed_args)):
+            pytest.skip(
```

Comment:
This should be a `ValueError` exception that stops the test execution

Reply:
Could also use pytest.UsageError here.
https://docs.pytest.org/en/latest/reference.html#usageerror